File: dataset/human/python/file_9605.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(200000)


def read():
    N, K = map(int, input().strip().split())
    A = list(map(int, input().strip().split()))
    return N, K, A


def solve(N, K, A, INF = 10**8):
    counts = [0 for i in range(N)]
    a = 1
    loop_init = -1
    loop_count = -1
    while True:
        counts[a-1] += 1
        if counts[a-1] == 2 and loop_count < 0:
            loop_count = 0
        elif counts[a-1] == 3:
            loop_count += 1
            break
        elif loop_count >= 0:
            loop_count += 1
        a = A[a-1]
    loop_init = sum([c == 1 for c in counts])

    a = 1
    if loop_init > K:
        for i in range(K):
            a = A[a-1]
    else:
        loop_k = (K - loop_init) % loop_count
        for i in range(loop_init + loop_k):
            a = A[a-1]
    return a


if __name__ == '__main__':
    inputs = read()
    outputs = solve(*inputs)
    if outputs is not None:
        print("%s" % str(outputs))

File: dataset/human/c/file_9605.c
#include <stdio.h>

int main(void){

  int n,d,x;
  scanf("%d %d %d", &n, &d, &x);
  int a[n];
	for (int i = 0; i < n; i++) {
		scanf("%d", &a[i]);
	}
	int sum = x;
	for (int i = 0; i < n; i++) {
		for (int j = 1; j <= d; j += a[i]) {
			sum++;
		}
	}
	printf("%d\n", sum);

  return 0;
}
